processor/attributesprocessor/attributes.go

```diff
@@ -126,39 +101,61 @@ func (a *attributesProcessor) Shutdown(context.Context) error {
 	return nil
 }
 
-func insertAttribute(action attributeAction, attributesMap map[string]*tracepb.AttributeValue) {
-	// Insert is only performed when the target key does not already exist
-	// in the attribute map.
-	if _, exists := attributesMap[action.Key]; exists {
+func (a *attributesProcessor) processSpan(span data.Span, serviceName string) {
+	if span.IsNil() {
+		// Do not create empty spans just to add attributes
 		return
 	}
 
-	setAttribute(action, attributesMap)
-}
-
-func updateAttribute(action attributeAction, attributesMap map[string]*tracepb.AttributeValue) {
-	// Update is only performed when the target key already exists in
-	// the attribute map.
-	if _, exists := attributesMap[action.Key]; !exists {
+	if a.skipSpan(span, serviceName) {
 		return
 	}
 
-	setAttribute(action, attributesMap)
+	attrs := span.Attributes()
+	for _, action := range a.config.actions {
+		// TODO https://github.com/open-telemetry/opentelemetry-collector/issues/296
+		// Do benchmark testing between having action be of type string vs integer.
+		// The reason is attributes processor will most likely be commonly used
+		// and could impact performance.
+		switch action.Action {
+		case DELETE:
+			attrs.Delete(action.Key)
+		case INSERT:
+			av, found := getSourceAttributeValue(action, attrs)
+			if !found {
```

Comment:
if `found` continue

Reply:
getSourceAttributeValue determines what value should be set for the "key" in the config:
* If the user specified a value in the cofig use that
* If the user specified a value from another attribute in the map, retrieve that.

Reply:
So if we don't found a value to apply in the operation we don't have anything to do.